core/domain/story_domain.py

```diff
@@ -0,0 +1,369 @@
+
+
+# Do not modify the values of these constants. This is to preserve backwards
+# compatibility with previous change dicts.
+STORY_PROPERTY_TITLE = 'title'
+STORY_PROPERTY_DESCRIPTION = 'description'
+STORY_PROPERTY_NOTES = 'notes'
+STORY_PROPERTY_LANGUAGE_CODE = 'language_code'
+
+STORY_NODE_PROPERTY_DESTINATION_NODE_IDS = 'destination_node_ids'
+STORY_NODE_PROPERTY_ACQUIRED_SKILL_IDS = 'acquired_skill_ids'
+STORY_NODE_PROPERTY_PREREQUISITE_SKILL_IDS = 'prerequisite_skill_ids'
+STORY_NODE_PROPERTY_OUTLINE = 'outline'
+STORY_NODE_PROPERTY_EXPLORATION_ID = 'exploration_id'
+
+
+# These take additional 'property_name' and 'new_value' parameters and,
+# optionally, 'old_value'.
+CMD_UPDATE_STORY_PROPERTY = 'update_story_property'
+CMD_UPDATE_STORY_NODE_PROPERTY = 'update_story_node_property'
+
+# These take node_id as parameter.
+CMD_ADD_STORY_NODE = 'add_story_node'
+CMD_DELETE_STORY_NODE = 'delete_story_node'
+
+
+class StoryChange(object):
+    """Domain object for changes made to story object."""
+    STORY_PROPERTIES = (
+        STORY_PROPERTY_TITLE, STORY_PROPERTY_DESCRIPTION,
+        STORY_PROPERTY_NOTES, STORY_PROPERTY_LANGUAGE_CODE)
+
+    STORY_NODE_PROPERTIES = (
+        STORY_NODE_PROPERTY_DESTINATION_NODE_IDS,
+        STORY_NODE_PROPERTY_ACQUIRED_SKILL_IDS,
+        STORY_NODE_PROPERTY_PREREQUISITE_SKILL_IDS, STORY_NODE_PROPERTY_OUTLINE,
+        STORY_NODE_PROPERTY_EXPLORATION_ID)
+
+    def __init__(self, change_dict):
+        """Initialize a StoryChange object from a dict.
+
+        Args:
+            change_dict: dict. Represents a command. It should have a 'cmd'
+                key, and one or more other keys. The keys depend on what the
+                value for 'cmd' is. The possible values for 'cmd' are listed
+                below, together with the other keys in the dict:
+                - 'update_story_property' (with property_name, new_value
+                and old_value)
+                - 'update_story_node_property' (with property_name, new_value
+                and old_value)
+
+        Raises:
+            Exception: The given change dict is not valid.
+        """
+        if 'cmd' not in change_dict:
+            raise Exception('Invalid change_dict: %s' % change_dict)
+        self.cmd = change_dict['cmd']
+
+        if self.cmd == CMD_ADD_STORY_NODE:
+            self.node_id = change_dict['node_id']
+        elif self.cmd == CMD_DELETE_STORY_NODE:
+            self.node_id = change_dict['node_id']
+        elif self.cmd == CMD_UPDATE_STORY_NODE_PROPERTY:
+            if (change_dict['property_name'] not in
+                    self.STORY_NODE_PROPERTIES):
+                raise Exception('Invalid change_dict: %s' % change_dict)
+            self.node_id = change_dict['node_id']
+            self.property_name = change_dict['property_name']
+            self.new_value = change_dict['new_value']
+            self.old_value = change_dict.get('old_value')
```

Comment:
We might have used .get() in the past because explorations existed before we introduced change lists. But now that we're building from scratch, as long as we are certain that old_value will always be present in the change list and that a KeyError should not arise with normal operation, let's just access it directly.

Ditto below (line 97).

Reply:
In exploration, old_value was given as optional, so I thought that could be done here also. Will change that.